crates/sargon/src/radix_connect/wallet_interaction/dapp_wallet_interaction/dapp_to_wallet/interaction_items/pre_authorization/expiration.rs

```diff
@@ -0,0 +1,66 @@
+use crate::prelude::*;
+
+/// An enum that represents the different ways a subintent can expire.
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, uniffi::Enum)]
+#[serde(tag = "discriminator", content = "value", rename_all = "camelCase")]
+pub enum DappToWalletInteractionSubintentExpiration {
+    /// The subintent expires at a specific time.
+    #[serde(rename = "expireAtTime")]
+    AtTime(Timestamp),
+
+    /// The subintent expires X seconds after its signature.
+    #[serde(rename = "expireAfterSignature")]
```

Comment:
not your model I guess, but this is really a strange name... signing is a 100% offline action. An Elliptic Curve Signature has no anchor in any system, alas this name makes it sound like it has. So I feel this is completely wrong.

Or have we built a system where Dapp receiving signed subitents will check the expiry and SHOULD skip using it if `AfterSignature.0` time has passed ? Some dapps can technically chose to ignore this right? Or is it done by our own tooling, in RDT?

Reply:
You are right that signing is a 100% offline action. It isn't actually expiring after the signature, but after the submission of the response (which has been signed). Considering the submission takes place after signing it, I don't have a big issue against the naming.


Regarding how it works, I don't think it is the dApp responsibility to check the expiration. It is just that the _parent transaction_ would fail if at least one of the subintents has already expired.

For example, a dApp sends a subintent for `User A` to approve sending 100 XRD with 1 hour expiration, and a subintent for `User B` to approve sending 2 USDT with same expiration.

If both users sign their subintents within one hour from each other, the transaction to exchange 100 XRD over 2 USDT will succeed. Otherwise, it would fail.

Note that with this expiration, the TX will be valid if both users decide to not check their wallet for N days, and open it to approve the subintent at the same time (with a 1 hour window). If the dApp wants to prevent this, it would use the other `DappToWalletInteractionSubintentExpiration::ExpireAtTime`. But in any case, it wouldn't be the dApps responsibility to check/ignore this: the TX would fail if any of the children doesn't meet the expiration.

Reply:
We should call this `SubmissionSyncWindowForParties` really. It is about submission, not signing, so we should use the word lemma "submit" not "sign". And it is a sync window for parties - as you described above, all parties have to sign within the that time window.

Reply:
and what you described should be put as doc on the variant :)

Reply:
AllSuubintentsSignedWithin

Reply:
well, to be honest, I prefer `Expiration` over `SubmissionSyncWindowForParties`. I would keep the enum as is, since it is also the name used on the JSON field. 

I believe the main confusion is on the `AfterSignature` variant. I would refer to the other subintents (`AllSubintents..`) since it is this subintent the one that will expire. Maybe we we can rename to `TimeAfterSubmission` instead?

Reply:
I mean to say that the enum should continue be named `Expiration` but this variant (case) `AfterSignature` ought to be renamed `AllSubintentsSubmittedWithin`